### PR TITLE
feat(testing-helpers): add fixture option to define wrapper el

### LINKED ISSUE #1125: [testing-helpers] Allow to inject wrapper element in fixture
Currently there's no way to customize the creation of wrapper element used as parent to the template.

A use case is testing an element that uses the dependency injection technique described by Polymer dev [here](https://www.youtube.com/watch?v=6o5zaKHedTE) and [here](https://youtu.be/x9YDQUJx2uw?t=2444). It relies on a parent element that listen to an event dispatched in the connectedCallback

An actual example of testing is [here](https://github.com/blikblum/lit-element-mobx-realworld-example-app/blob/master/test/app-header.js#L23). `app-header` element depends on a context provided by a parent element. Currently there's no way to properly test it.

This is something i can work if approved

### PR DESCRIPTION
**Pull request by blikblum**
Fixes #1125 

This is a quick PR to show how the feature could be implemented (mith minimal changes and backward compatible)

The option name could change from `wrapper` top `parent`

### CODE REVIEW

**Comment by blikblum** [comment]
Renamed option from `wrapper` to `parent` since the later is shorter and easier to remember

File: packages/testing-helpers/src/fixtureWrapper.js [hunk 1]
```diff
@@ -7,8 +7,7 @@ export const cachedWrappers = [];
  *
  * @returns {Element}
  */
-export function fixtureWrapper() {
-  const wrapper = document.createElement('div');
+export function fixtureWrapper(wrapper = document.createElement('div')) {
```
Comment (bennypowers) [inline packages/testing-helpers/src/fixtureWrapper.js]: nit: for consistency, rename to `parent`

File: packages/testing-helpers/src/fixture-no-side-effect.js [hunk 2]
```diff
@@ -10,14 +10,15 @@ import { isValidRenderArg } from './lib.js';
  *
  * @template {Element} T
  * @param {import('./litFixture').LitHTMLRenderable} template Either a string or lit-html TemplateResult
+ * @param {object} [options]
```
Comment (bennypowers) [inline packages/testing-helpers/src/fixture-no-side-effect.js]: Please define a `FixtureOptions` typedef

File: packages/testing-helpers/src/stringFixture.js [hunk 3]
```diff
@@ -7,10 +7,11 @@ import { elementUpdated } from './elementUpdated.js';
  *
  * @template {Element} T - Is an element or a node
  * @param {string} template
+ * @param {object} [options]
  * @returns {T}
  */
-export function stringFixtureSync(template) {
-  const wrapper = fixtureWrapper();
+export function stringFixtureSync(template, options = {}) {
+  const wrapper = fixtureWrapper(options.parent);
```
Comment (bennypowers) [inline packages/testing-helpers/src/stringFixture.js]: nit: Please rename here to parent

Comment (blikblum) [reply packages/testing-helpers/src/stringFixture.js]: Done

File: packages/testing-helpers/src/fixture-no-side-effect.js (hunk 2 above)
Comment (blikblum) [reply packages/testing-helpers/src/fixture-no-side-effect.js]: Done

**Comment by blikblum** [comment]
@bennypowers i did the changes as your review. Is there anything else i should do?

**Comment by daKmoR** [comment]
potentially open todos
- rename `parent` to `parentNode` to make it clear that it's a node
- add "nicer" jsDocs for the options to improve autocomplete docs

PS: @bennypowers is in charge of this change - he will make the change himself when he's available, you're welcome to push as well. thank you for your patience 🤗

**Comment by bennypowers** [comment]
We're on the home stretch

**Comment by blikblum** [comment]
Manu thanks

**Pull request merged**